### DISCUSSION TITLE
Play framework migration

### DISCUSSION THREAD

**Discussion by nehapatkar14** [playframework/playframework#11531] [Q&A]
Hi , I have my application wars build on java 8 , play 2.3 .this application works fine with java 15 .However when the same wars are deployed to work with java 17 it gives me a distorted UI. Can any on explain me what is different when the same wars work with java 17?

**Comment by mkurz** [comment]
So you say you run a Play 2.3 app with Java 15 and now want to upgrade to Java 17?
I doubt that's a good idea. We just recently supports Java 17 starting [with Ply 2.8.15](https://github.com/playframework/playframework/releases/tag/2.8.15), so I don't think that will work with such an outdated Play 2.3 version.
You should upgrade to Play 2.8.x as soon as possible.

**Reply by nehapatkar14** [reply]
My application war is built with java 8 and play 2.3 .When I deploy this war to work with java 15 it works fine , but with java 17 it gives distorted UI.

**Reply by nehapatkar14** [reply]
Does play 2.8 work with java 17 ? when I tried to build a simple hello world program  with java 17  the localhost:9000 gave error , but the same worked with java 8

**Reply by mkurz** [reply]
Like I said, Play 2.3 does not support Java 17. Maye you were lucky it worked with Java 15, that depends what features of the frameworks you are using. I highly recomment upgrading to latest Play 2.8.x, which, from also is highly recommended from a safety point of view (CVEs). You should start with https://www.playframework.com/documentation/2.8.x/Migration24 and work your way up to 2.8.x, also always check the highlights of each release (e.g. for 2.4.x https://www.playframework.com/documentation/2.8.x/Highlights24)

**Reply by mkurz** [reply]
Yes Java 17 is working in Play 2.8 but you need to update some libraries in your `build.sbt`, see: https://github.com/playframework/playframework/releases/tag/2.8.15

**Reply by nehapatkar14** [reply]
Can you point me out to some example showing how to use internationalization of messages with play 2.8 using java controllers

**Reply by mkurz** [reply]
> Can you point me out to some example showing how to use internationalization of messages with play 2.8 using java controllers

I think you should find all you need in https://www.playframework.com/documentation/2.8.x/JavaI18N

**Reply by nehapatkar14** [reply]
will play 2.6 support java 17?

**Reply by mkurz** [reply]
No I doubt it will work with Play 2.6

**Reply by nehapatkar14** [reply]
Thank you so much , I am really struggling with this internationalization of messages to implement in my project earlier using 2.3 and migrating it to 2.8 . If there is any sample project demonstrating Internationalization with play 2.8 which I can download and just run to see how it works it would be a big help . Thank you so much for your support !!

**Reply by mkurz** [reply]
Have a look at https://github.com/playframework/play-samples and `grep` from "messages": https://github.com/playframework/play-samples/search?q=messages maybe that helps.

**Reply by nehapatkar14** [reply]
Sure thank you so much for your help and support!!
